src/tokenizer.rs

```diff
@@ -396,14 +398,59 @@ fn parse_string_literal<'a>(current: &'a str, current_position: &SourcePosition)
     Ok((advance_result, TokenKind::StringLiteral(literal)))
 }
 
-fn parse_long_string_literal<'a>(current: &'a str, current_position: &SourcePosition) -> Result<(AdvanceResult<'a>, TokenKind<'a>), AdvanceError> {
+fn parse_multi_line_string_literal<'a>(current: &'a str, current_position: &SourcePosition) -> Result<(AdvanceResult<'a>, TokenKind<'a>), AdvanceError> {
+    match parse_multi_line_thing(current, &current_position) {
+        Err(e) => Err(e),
+        Ok((advance_result, raw_content, depth)) =>
+            Ok((advance_result, TokenKind::StringLiteral(
+                StringLiteral::LongForm { raw_content, depth },
+            )))
+    }
+}
+
+fn parse_multi_line_thing<'a>(current: &'a str, current_position: &SourcePosition) -> Result<(AdvanceResult<'a>, Cow<'a, str>, u32), AdvanceError> {
+    let captures = match PATTERN_MULTI_LINE_STRING_START.captures(current) {
+        Some(c) => c,
+        None => return Err(AdvanceError::NoMatch),
+    };
```

Comment:
We can use `.ok_or(AdvanceError::NoMatch)?` here instead, which is more concise than the explicit match.